### ISSUE TITLE
Requests inside phpunit tests always have error response (Lumen 5.3)

### ISSUE THREAD

**Issue by boyd91** [dingo/api#1243]
Whenever I try to test an API endpoint in my Lumen 5.3 setup the requests response has the error  `The version given was unknown or has no registered routes.`.

I seems to occur with all kind of requests executed in tests.

It tested the following request methods, which all have the same error response
```php
$this->post();
$this->get();
$this->put();
$this->JSON();
```

Here's one complete test
```php
use Laravel\Lumen\Testing\DatabaseTransactions;

class AdministrationsTest extends TestCase
{

    use DatabaseTransactions;

    public function test_create_administrations()
    {
        $response = $this->post('api/administrations', [
            'label' => 'Administration 1'
        ], ['Accept' => 'application/json']);

        // The error is in the response here
        dd($response->response->getContent()); 
        
        $this->seeInDatabase('administrations', ['label' => 'Administration 1']);
    }
}
```

Here's my phpunit.xml
```xml
<?xml version="1.0" encoding="UTF-8"?>
<phpunit backupGlobals="false"
         backupStaticAttributes="false"
         bootstrap="bootstrap/app.php"
         colors="true"
         convertErrorsToExceptions="true"
         convertNoticesToExceptions="true"
         convertWarningsToExceptions="true"
         processIsolation="false"
         stopOnFailure="false"
         syntaxCheck="false">
    <testsuites>
        <testsuite name="Application Test Suite">
            <directory>./tests/</directory>
        </testsuite>
    </testsuites>
    <filter>
        <whitelist>
            <directory suffix=".php">app/</directory>
        </whitelist>
    </filter>
    <php>
        <env name="APP_ENV" value="testing"/>
        <env name="CACHE_DRIVER" value="array"/>
        <env name="API_PREFIX" value="api"/>
        <env name="API_DOMAIN" value="*censored*"/>
        <env name="API_DEBUG" value="true"/>
        <env name="API_STRICT" value="false"/>
        <env name="API_DEFAULT_FORMAT" value="json"/>
    </php>
</phpunit>
```

**Comment by boyd91** [comment]
So I did some digging inside the package (master @ dc312a5). I wasn't able to solve the issue but found some additional info that might be relevant.

1. The exception is thrown here https://github.com/dingo/api/blob/master/src/Routing/Adapter/Lumen.php#L90

    1. Inside that method the argument `$version` is `string(2) "v1"`

    2. Inside that method `$this->routes` is an *empty* array

2. The addRoute method on the same class is called https://github.com/dingo/api/blob/master/src/Routing/Adapter/Lumen.php#L153

    1. Inside that method the argument `$versions` is an array containing the version. `[ 0 => 'v1' ]`

    2. The method *does* succesfully add the route to `$this->routes` with the correct version array-key.

    3. I tested if after any time the `addRoute()` method was called `$this->routes` would be empty. But it wasn't. So the logic inside `addRoute()` does is not the one clearing `$this->routes`

**Conclusion**
When the route is added the internal route collection is properly populated. But when some route gets dispatched `$this->routes` is empty, which causes the issue.

**I debugged within the phpunit environment by outputting the debug log messages when running phpunit.**

P.S. For clearity I should add that my issue occurs only when running phpunit tests and not when I query my application over HTTP.

**Comment by boyd91** [comment]
**I found the source of the issue**
My routes were located inside a different files than Lumens default `routes.php`. I had the other file `required_once` inside the default `routes.php`. When I placed all my files inside the default one it suddenly worked.

**So a different routes file included with `require_once` becomes a problem when you make requests in phpunit tests but not when you make requests over HTTP.**

Although I think it should still be fixed, my specific issue is resolved now so I'm closing it.
